rust/definitions/src/navigation.rs

```diff
@@ -0,0 +1,661 @@
+use crate::{
+    crypto::Encryption, history::Event, keyring::NetworkSpecsKey, network_specs::NetworkSpecs,
+};
+
+pub use crate::network_specs::NetworkSpecsToSend;
+
+#[derive(PartialEq, Clone)]
+pub struct SeedNameWithIdenticon {
+    pub seed_name: String,
+    pub identicon: Vec<u8>,
+}
+
+/// Enum containing card sets for three different outcomes:
+/// signing (Sign), accepting (Stub) and reading, for example, in case of an error (Read)
+#[derive(PartialEq, Debug, Clone)]
+pub enum TransactionAction {
+    Derivations {
+        content: TransactionCardSet,
+        network_info: NetworkSpecs,
+        checksum: u32,
+        network_specs_key: NetworkSpecsKey,
+    },
+    Sign {
+        content: TransactionCardSet,
+        checksum: u32,
+        has_pwd: bool,
+        author_info: Address,
+        network_info: NetworkSpecs,
+    },
+    Stub {
+        s: TransactionCardSet,
+        u: u32,
+        stub: StubNav,
+    },
+    Read {
+        r: TransactionCardSet,
+    },
+}
+
+/// Enum describing Stub content.
+/// Is used for proper navigation. Variants:
+/// AddSpecs (with associated NetworkSpecsKey), LoadMeta (with associated
+/// NetworkSpecsKey for the first by order network using those metadata),
+/// and LoadTypes
+#[derive(PartialEq, Debug, Clone)]
+pub enum StubNav {
+    AddSpecs { n: NetworkSpecsKey },
+    LoadMeta { l: NetworkSpecsKey },
+    LoadTypes,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum FooterButton {
+    Log,
+    Scan,
+    Keys,
+    Settings,
+    Back,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum RightButton {
+    LogRight,
+    NewSeed,
+    Backup,
+    MultiSelect,
+    NDMenu,
+    TypesInfo,
+    KeyMenu,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ScreenNameType {
+    H1,
+    H4,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ShieldAlert {
+    Active,
+    Past,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum AlertData {
+    Shield { f: Option<ShieldAlert> },
+    ErrorData { f: String },
+    Confirm,
```

Comment:
Here the idea was to control all confirmation message from single place. Alert data should have following structure then:

```
struct Contirm {
  line1: Option<String>
  line2: Option<String>
  yes_message: Option<String>
  no_message: Option<String>
}
```
Thus UI will default to something like 

>CONFIRM
>Are you sure?
>yes/no

AFAIR this is not really implemented on Rust side yet and many of these alerts are still called by UI. We should clearly change soon that because bugs in those are potential security-related bugs. Not sure if we should already start preparing for this, please create an issue so that we do not forget if postponing.